src/saltext/vmware/utils/license_mgr.py

```diff
@@ -0,0 +1,448 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+# pylint: disable=no-name-in-module
+import logging
+
+# noreorder
+import salt.exceptions
+import salt.modules.cmdmod
+import salt.utils.path
+import salt.utils.platform
+import salt.utils.stringutils
+import saltext.vmware.utils.cluster as utils_cluster
+import saltext.vmware.utils.datacenter as utils_datacenter
+import saltext.vmware.utils.esxi as utils_esxi
+from saltext.vmware.utils.common import get_service_content as get_service_content
+
+try:
+    from pyVmomi import vim, vmodl
+
+    HAS_PYVMOMI = True
+except ImportError:
+    HAS_PYVMOMI = False
+
+log = logging.getLogger(__name__)
+
+
+def _list_lic_keys(licenses):
+    """
+    Return a list of license keys
+
+    licenses:
+       vCenter licenses as made available from licenseManager
+
+    Note:
+        skips evaluation licenses
+    """
+    log.debug("started _list_lic_keys")
+    lic_keys = []
+    for lic in licenses:
+        if lic.used is None:
+            continue
+        lic_keys.append(lic.licenseKey)
+    return lic_keys
+
+
+def _find_lic_for_key(licenses, license_key):
+    """
+    Return license which matches specified license_key
+
+    licenses:
+       vCenter licenses as made available from licenseManager
+
+    license_key
+        license key to search for
+
+    Note:
+        skips evaluation licenses
+    """
+    log.debug("started _find_lic_for_key")
+    lic_keys = []
+    for lic in licenses:
+        if lic.used is None:
+            continue
+        if license_key == lic.licenseKey:
+            return lic
+    return None
+
+
+def is_vcenter(service_instance):
+    """
+    Test if service_instance represents vCenter,
+    otherwise assume represents an ESXi server
+
+    service_instance
+        The Service Instance to check if it represents a vCenter
+
+    Return:
+        True  - vCenter
+        False - ESXi server
+        None - neither of the above
+    """
```

Comment:
The original Ansiable (v2.7) allowed for licensing vCenter or ESXi, so boolean seemed appropriate.

The None was added at the start of development in case I needed a case where I needed to handle some weird situation.  Which I never ran into. Wondering about handling Datacenter and Clusters now ?  But its usage is really to ensure working on vCenter, hence the boolean True is fine

Reply:
Actually going to change this since really only usage and testing is 'am I on a vCenter', which simple boolean makes much more sense, will do it Monday